infinite_tracing/lib/infinite_tracing/channel.rb

```diff
@@ -5,8 +5,8 @@
 module NewRelic::Agent
   module InfiniteTracing
     class Channel
-      COMPRESSION_LEVELS = %i[none low medium high].freeze
-      DEFAULT_COMPRESSION_LEVEL = :none
+      SETTINGS_BASE = {'grpc.enable_deadline_checking' => 0}.freeze
+      SETTINGS_COMPRESSION_DISABLED = SETTINGS_BASE.merge({'grpc.minimal_stack' => 1}).freeze
```

Comment:
Most of the content removed from this `channel.rb` file was simply moved to `config.rb`, as it is more appropriate to be considered configuration related.

Before this PR, everyone got both `enable_deadline_checking' => 0` and `minimal_stack' => 1`. Now `minimal_stack => 1` is not used when compression is desired, as we have learned it will disable compression.